tools/benchmark/src/main/java/com/datadog/benchmark/internal/DatadogMetricProfiler.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.benchmark.internal
+
+import android.view.Choreographer
+import com.datadog.benchmark.ext.canReadSafe
+import com.datadog.benchmark.ext.existsSafe
+import com.datadog.benchmark.ext.readLinesSafe
+import com.datadog.benchmark.ext.readTextSafe
+import java.io.File
+import java.util.concurrent.TimeUnit
+
+internal class DatadogMetricProfiler {
+
+    private var currentFps: Int = 0
+    private var frameCount = 0
+    private var lastFrameTime: Long = 0
+    private val intervalMs = FPS_SAMPLE_INTERVAL_IN_MS
+
+    private val statusFile = STATUS_FILE
+    private val statFile = STAT_FILE
+
+    private var lastCpuTicks = 0L
+
+    private val frameCallback = object : Choreographer.FrameCallback {
+        override fun doFrame(frameTimeNanos: Long) {
+            if (lastFrameTime == 0L) {
+                lastFrameTime = System.nanoTime()
+            }
+
+            frameCount++
+            val currentFrameTime = System.nanoTime()
+            val elapsedTime: Long = currentFrameTime - lastFrameTime
+
+            if (elapsedTime >= TimeUnit.MILLISECONDS.toNanos(intervalMs)) {
+                val fps: Double = frameCount / (elapsedTime / NANO_IN_SECOND)
+                currentFps = fps.toInt()
+                lastFrameTime = currentFrameTime
+                frameCount = 0
+            }
+            Choreographer.getInstance().postFrameCallback(this)
```

Comment:
why Choreographer is chosen over what `androidx.metrics` gives?

Reply:
I didn't know `androidx.metrics` can do the same thing, is there any advantage of `android.metrics` comparing to  `Choreographer`?

Reply:
yes, a lot of advantages: it a wrapper around different new APIs allowing collecting more granular metrics. Also by registering `Choreographer` callback and not un-registering it when app goes into background/losing focus, you make CPU running more (this point probably doesn't matter a lot for internal tooling).

Anyway, FPS is not the good metric to judge about the performance and we were discussion to ditch it and replace it with something else (especially considering dynamic refresh rate). Much more meaningful metric will be something like the % of skipped/junky frames, for example.

Reply:
Good to know, I will first make a pull request to clean up the code on all the places where you mentioned, and another dedicated commit to replace the `Choreographer`

Reply:
Overall maybe for this work we shouldn't use FPS as a metric at all. But this is just my opinion.

Reply:
Maybe you are right, with current synthetics test and metrics result, we can hardly see any difference between baseline and instrumented test, I will try to see if `androidx.metrics` can provide some more valuable result.